core/src/main/java/org/wso2/msf4j/internal/router/MicroserviceMetadata.java

```diff
@@ -145,7 +142,7 @@ private boolean isHttpMethodAvailable(Method method) {
                                                                          List<String> acceptHeader)
             throws HandlerException {
         try {
-            String path = URI.create(uri).normalize().getPath();
+            String path = Utils.normalizePath(URI.create(uri).normalize().getPath());
```

Comment:
We do not need to normalise the path here since we already do normalise in `patternRouter.getDestinations` method. We should ideally also normalise the path in `getMatchedDestination` method in `MicroserviceMetaData` class (added a comment separately regarding this). WDYT?

Reply:
Decided to convert code that did open-coded "/" removal with calls to `normalizePath`.  That is done now.